app/blueprints/assessments/routes.py

```diff
@@ -1629,6 +1646,54 @@ def application(application_id):
             )
         )
 
+    thread_executor = ContextAwareExecutor(
```

Comment:
I'm not sure the savings from the thread executor mechanism are worth the complexity or if this logic shouldn't be encapsulated lower down. I know this logic was added before this PR so all good for it to go through.

I wonder if using the `async` functionality added to flask recently (relatively!) would be a less hands on way of managing waiting for multiple calls.